### DISCUSSION TITLE
Converting GUID and Metadata to readable text

### DISCUSSION THREAD

**Discussion by KishanMistrya8** [thoughtspot/cs_tools#109] [Q&A]
Hi,

One of our clients would like a list of all liveboards in their ThoughtSpot system along with the date it was created and the owner.

They provided me with these commands:

1. atlascli --adv

2. node search pinboard --num 10000 --file /home/admin/liveboard
 
This allowed me to get a list of all the liveboards, but as seen in the image attached, the date and owner column are full of GUID string and metadata. Is there a way to convert this into readable text?

![image](https://github.com/thoughtspot/cs_tools/assets/141828357/bfc65f7e-123c-4c65-8800-15a7327da51d)

If anyone could help with this, that would be much appreciated.

**Comment by boonhapus** [comment]
Hi @KishanMistrya8 , this data comes in the same format out of the Searchable tool. It pulls from the Metadata APIs in ThoughtSpot, which also are powered by the service `atlas` internally. I would recommend running Searchable to a CSV file and using the data from there.

```bash
cs_tools tools searchable gather --syncer csv://directory=.
```

You'll get both `TS_METADATA_OBJECT` and `TS_USER` files. Objects CSV can be filtered by type `PINBOARD_ANSWER_BOOK` to find the Liveboards, and each row will contain an `AUTHOR_GUID` which can be matched to the user guid in `TS_USER`.

**Reply by KishanMistrya8** [reply]
Hi boonhapus,

Thanks for your reply,

Will cs_tools need to be installed on the server to run this command, as at the moment I am getting "-bash: cs_tools: command not found"

If so, how would I go about installing cs tools, and will it require any downtime?

Thanks,
Kishan

**Reply by boonhapus** [reply]
No downtime is necessary. CS Tools is not an official ThoughtSpot product, but you are free to install it anywhere which has network connectivity to your ThoughtSpot instance. We use the ThoughtSpot APIs to build most of these tools.

You can find install and config documentation here:
https://thoughtspot.github.io/cs_tools/tutorial/install/#first-time-install

**Reply by KishanMistrya8** [reply]
Hi boonhaous,

Is it possible to do this from Windows powershell or do I have to install CS tools on the Thoughtspot cluster? When I click on the Thoughtspot download, it says "We strongly recommend against this option"

If it is possible to do this from Windows powershell, how would I go about this?

Thanks,
Kishan

**Reply by boonhapus** [reply]
@KishanMistrya8 It's best if you DON'T install directly on the cluster. To get going from Windows, you can try executing this from powershell.

```powershell
(Invoke-WebRequest `
    -Uri https://raw.githubusercontent.com/thoughtspot/cs_tools/master/cs_tools/updater/_bootstrapper.py `
    -UseBasicParsing `
).Content | python - --install
```

**Reply by KishanMistrya8** [reply]
Hi boonhapus,

Once installed, what would I have to do?

How do I get the data into powershell?

**Reply by boonhapus** [reply]
@KishanMistrya8 you'll want to [**create a configuration**](https://thoughtspot.github.io/cs_tools/tutorial/config/) (so you can connect to your ThoughtSpot instance) and then run the below command

`cs_tools tools searchable gather --syncer csv://directory=.`

This will create many CSV files which you can use to look up the data you need. `ts_metadata_object.csv` and `ts_user.csv` specifically!
